pkg/deviceplugin/server.go

```diff
@@ -163,8 +166,9 @@ func (srv *server) Update(devices map[string]DeviceInfo) {
 func (srv *server) setupAndServe(namespace string, devicePluginPath string, kubeletSocket string) error {
 	resourceName := namespace + "/" + srv.devType
 	pluginPrefix := namespace + "-" + srv.devType
+	restart := true
 
-	for {
+	for restart {
```

Comment:
Thanks! That's a good catch.

I think this line adds a new state to the `server`. How about making it more explicit with introducing `server.state` which could be `serving` and `terminating`? Later `uninitialized` can be added for better control of state transitions.

Then in `server.Stop()` we could just set the state to `terminating` and remove the socket file without opening additional channels.

Reply:
Sure, that's much cleaner, thank you.